frontend/src/components/TagFilter/index.tsx

```diff
@@ -0,0 +1,220 @@
+import React, {useEffect, useState}  from 'react';
+import styled from 'styled-components';
+import axios from 'axios';
+import { Typography, Checkbox, FormControlLabel, Button, Tabs, Tab, Box } from '@material-ui/core';
+import SearchResultsInfiniteScroll from '../../components/SearchResultsInfiniteScroll';
+
+const TypeSelector = styled.div`
+    display: flex;
+    justify-content: space-between;
+`;
+
+const TagFilter: React.FC = () => {
+    const [allTags, setTags] = useState<Array<any>>([]);
+    const [isMovieSelected, selectMovie] = useState<Boolean>(false);
+    const [isBookSelected, selectBook] = useState<Boolean>(false);
+    const [allSelectedTags, selectTags] = useState<Array<any>>([]);
+    const increaseIndexBy = 5;
+    const [queryStartIndex, setQueryStartIndex] = useState(0);
+    const [hasMoreMovieResults, setHasMoreMovieResults] = useState<boolean>(true);
+    const [hasMoreBookResults, setHasMoreBookResults] = useState<boolean>(true);
+    const [allResults, setAllResults] = useState<Array<any>>([]);
+
+    useEffect(() => {
+        axios.get('/nextStoryTags')
+            .then((res: any) => {
```

Comment:
you don't need to sort the tags in the frontend anymore, that's done in the backend